### DISCUSSION TITLE
Alternatives for `rstudioapi::askForPassword()`

### DISCUSSION THREAD

**Discussion by albersonmiranda** [REditorSupport/vscode-R#793] [Q&A]
`rstudioapi::askForPassword()` is a good option when user must insert, say, a database password in the session. But it asks specifically for RStudio. If you try to run it in VSCode, then:

```
r$> rstudioapi::askForPassword("Database password")
Error in as.environment(match("tools:rstudio", search())) : 
  invalid 'pos' argument
```

So, is there an alternative that we can use or should we develop one?

**Comment by renkun-ken** [comment]
@MilesMcBain Looks like we could implement `rstudioapi::askForPassword()` which triggers a user input for password.

**Reply by coatless** [reply]
Glancing through the documentation, I didn't see the feature present. Is this still planned? It would be great to have the ability to shield sensitive information being sent.
